vassal-app/src/main/java/VASSAL/build/module/chessclockcontrol/ChessClock.java

```diff
@@ -0,0 +1,826 @@
+    PlayerRoster r = GameModule.getGameModule().getPlayerRoster();
+    ChessClockControl ctrl = ChessClockControl.getInstance();
+    if ((r != null) && (ctrl != null)) {
+      for (String s : r.getSides()) {
+        boolean okay = true;
+        for (ChessClock c : ctrl.getChessClocks()) {
+          if (c.getSide().equals(s)) {
+            okay = false;
+            break;
+          }
+        }
+        if (!okay) {
+          continue;
+        }
+        side       = s;
+        buttonText = s;
+        break;
+      }
+    }
+
+    setConfigureName(side);
+
+    image        = "";
+    elapsedTime  = 0;
+    verifiedTime = 0;
+    clockTicking = false;
+
+    // Timer is static so it gets created exactly once ever
+    if (timer == null) {
+      timer = new Timer(100, this);
+      timer.stop();
+    }
+
+    // But every ChessClock instance needs its own action listener
+    timer.addActionListener(this);
+
+    // Action listener for clicking on the clock
+    ActionListener al = e -> {
+      Command command = new NullCommand();
+      ChessClockControl ccc = ChessClockControl.getInstance();
+      if ((ccc != null) && ccc.getClocksTicking() > 0) {
+        command = command.append(ChessClockControl.getInstance().startNextClock());
+      }
+      else {
+        command = command.append(updateState(true));
+      }
+      command.execute();
+      GameModule.getGameModule().sendAndLog(command);
+
+      if (timer == null) {
+        setup(true);
+      }
+      startTimer();
+    };
+
+    // Create our actual timer button
+    timerButton = new LaunchButton(buttonText + " " + "0:00:00", TOOLTIP, null, null, ICON, al); //$NON-NLS-1$
+    timerButton.setFont(new Font("SansSerif", Font.BOLD, 12)); //$NON-NLS-1$
+    initTimerButton();
+  }
+
+  /**
+   * @return side this clock is assigned to
+   */
+  public String getSide() {
+    return side;
+  }
+
+  /**
+   * @return true if this clock is running
+   */
+  public boolean isTicking() {
+    return clockTicking;
+  }
+
+  /**
+   * @return elapsed time from local perspective
+   */
+  public long getElapsed() {
+    return elapsedTime;
+  }
+
+  /**
+   * @return elapsed time verified by handshake (or fiat, depending on situation)
+   */
+  public long getVerified() {
+    return verifiedTime;
+  }
+
+  /**
+   * @param name side to check if a referee
+   * @return True if passed side is a referee
+   */
+  protected static boolean isReferee(String name) {
+    return PlayerRoster.isSoloSide(name);
+  }
+
+  /**
+   *  Make sure the static millisecond timer is actually running (unless it hasn't even been initialized yet)
+   */
+  protected void startTimer() {
+    if (!timer.isRunning()) {
+      timer.start();
+    }
+  }
+
+  /**
+   *  Defines the Command by which information about this clock is communicated between machines (and to/from save and log files)
+   */
+  public class UpdateTimerCommand extends Command {
```

Comment:
Does it need to be public?

Reply:
Can it be static? Inner classes should be static whenever they can.

Reply:
apparently it can be private, so it is. Changing it to static made lots of things not compile.